packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,179 @@
+import { includes, find } from "lodash";
+import { XMLAttribute } from "@xml-tools/ast";
+import {
+  UI5SemanticModel,
+  UI5Aggregation,
+  UI5Association,
+  UI5Event,
+  UI5Prop,
+} from "@ui5-language-assistant/semantic-model-types";
+import {
+  getUI5ClassByXMLElement,
+  getUI5AggregationByXMLElement,
+  isXMLNamespaceKey,
+  flattenProperties,
+  flattenEvents,
+  flattenAssociations,
+  flattenAggregations,
+  ui5NodeToFQN,
+} from "@ui5-language-assistant/logic-utils";
+import { UnknownAttributeKeyIssue } from "../../../api";
+import { TEMPLATING_NS, CUSTOM_DATA_NS } from "../../utils/special-namespaces";
+
+export function validateUnknownAttributeKey(
```

Comment:
Subjective / Style question:

There are many branches which **return**, does it make sense to add more **else if** to clearly distinguish the logical flows instead of relying on needing to recognize the **return** statements?

It won't affect correctness, but would it make the flow easier to grok?

Reply:
That depends if you prefer `else if`s or smaller nesting levels. I'm used to smaller nesting levels...

Reply:
It style thing so your pick, I am slightly worried about refactoring and moving things around breaking correctness which is more obvious when there is an explicit if/else